Add context to behat which can check the contents with regexp

miya0001 opened the pull request:
I added the new context to behat which can check the file contents with regex like following.

```
When I run `echo "hello" > hello.txt`
Then the hello.txt file should match /^H.+o$/i
```

Related:
https://github.com/wp-cli/scaffold-command/pull/61#discussion_r143706183

danielbachhuber commented:
@miya0001 Although it's not obvious, the source of truth (e.g. original copy) for the test suite is [wp-cli/wp-cli](https://github.com/wp-cli/wp-cli/tree/master/features). Can you submit these same changes to that repo too?

danielbachhuber commented on features/steps/then.php:
```diff
@@ -199,6 +199,20 @@ function ( $world, $path, $type, $action, $expected = null ) {
 	}
 );
 
+$steps->Then( '/^the (.+) file should match (((\/.+\/)|(#.+#))([a-z]+)?)$/',
```
Actually, to make this a bit more explicit, could the syntax be:

```
the contents of the <file> file should match
```

miya0001 commented:
@danielbachhuber 

Fixed it and I added same context for STDOUT/STDERR.

Example:
```
Feature: Regex context

  Scenario: Scaffold a WP-CLI command without tests
    Given an empty directory

    When I run `echo "hello" > hello.txt`
    Then the contents of the hello.txt file should match /H.+o/i

    When I run `echo "hello"`
    Then STDOUT should match /H.+o/i
```
